### ISSUE TITLE
getting an error on the default service account parameter with 10.0.2 version of the project factory module

### ISSUE THREAD

**Issue by madhuakiti** [terraform-google-modules/terraform-google-project-factory#912]
### TL;DR

As I was trying to upgrade the module version from 9.2.1 to 10.0.2, I was getting an error related to the default service account action parameter. Based on the documentation from the 10.0.2 version this shouldn't happening.

Error: expected action to be one of [DEPRIVILEGE DELETE DISABLE], got KEEP

  on .terraform/modules/project/modules/core_project_factory/main.tf line 121, in resource "google_project_default_service_accounts" "default_service_accounts":
 121:   action         = upper(var.default_service_account)

### Expected behavior

Terragrunt/terraform plan should work as expected and should be able to print the plan output

### Observed behavior

Error: expected action to be one of [DEPRIVILEGE DELETE DISABLE], got KEEP

  on .terraform/modules/project/modules/core_project_factory/main.tf line 121, in resource "google_project_default_service_accounts" "default_service_accounts":
 121:   action         = upper(var.default_service_account)

### Terraform Configuration

```hcl
We are using the combination of terraform and terragrunt to managed our GCP landing zone or foundation
```

### Terraform Version

```sh
Current terraform version: 10.3.7
Current terragrunt version: 0.25.0
```
